src/livestreamer_cli/main.py

```diff
@@ -444,9 +449,12 @@ def format_valid_streams(streams):
             joined = delimiter.join(synonyms)
             name = "{0} ({1})".format(name, joined)
 
-        validstreams.append(name)
+        validstreams.append((name, weight))
+
+    validstreams.sort(key=lambda stream: stream[1])
```

Comment:
We alredy sort the streams [here](https://github.com/chrippa/livestreamer/blob/develop/src/livestreamer_cli/main.py#L436), so instead of sorting twice you could provide a key function there instead.